### ISSUE TITLE
Unable to upgrade to Netmiko 4.3.0 due to nornir-netmiko specific requirement on TextFSM v1.1.2

### ISSUE THREAD

**Issue by fseesink** [ktbyers/nornir_netmiko#66]
We currently leverage both `netmiko` directly as well as Netmiko via Nornir using the `nornir-netmiko` plugin in some of our projects.  And we keep hitting on dependency challenges wherein we cannot upgrade the former due to the latter.

For example, at present we are limited to Netmiko 4.1.2, released in Aug 2022, (4.2.0 was released in May, and current 4.3.0 was released 3 weeks ago), all because this plugin has a very specific requirement for TextFSM v1.1.2.  Is there any way to bring these projects into alignment?

**Comment by ktbyers** [comment]
...that being said, I will see if I can release a new version of `nornir_netmiko` in the next few days.

**Comment by ktbyers** [comment]
nornir-netmiko 1.0.1 just released.

https://github.com/ktbyers/nornir_netmiko/releases/tag/1.0.1

https://pypi.org/project/nornir-netmiko/1.0.1/

**Comment by fseesink** [comment]
> You should be able to just install nornir_netmiko and then netmiko 4.3.0
> 
> ```
> pip install nornir_netmiko
> pip install netmiko==4.3.0
> ```
> 
> pip will complain, but it will install everything.
> 
> ```
> [netmiko_packaging] ktbyers@pydev2 ~/VENV 
> $ pip list | grep netmiko
> netmiko        4.3.0
> nornir-netmiko 1.0.0
> 
> [netmiko_packaging] ktbyers@pydev2 ~/VENV 
> $ pip list | grep textfsm
> textfsm        1.1.3
> ```
> 
> There isn't anything blocking you from using the latest Netmiko.
> 
> Yes, at one point we had to pin various things to textfsm==1.1.2 (as there were package/release issues in the TextFSM project) that were subsequently fixed.

Yes I realize I could have done that.  In fact, it was by doing that that I found the pinned dependency.  This isn't the first time this has happened with this particular set of modules.

But usually when a Python package has a pinned version, there is (or should be) a reason for it.  And I wasn't about to take that chance.  So I rolled back to netmiko 4.1.2, which appears to have been the last supported version.

And I realize the joys of trying to keep the dependency chain down can be... fun.  (It's one of the reasons I have come to really like Go, as a lot of those headaches just go away.  But that's neither here nor there in this context.)  But usually when coding in Python, at least if you leverage usual virtual environs, the worst case tends to be a single level dependency.  This just happens to be more annoying in that if you want to use Netmiko to do direct sessions to devices, but then you also want to leverage Nornir in some cases, which in turn relies on nornir-netmiko, you get into this dependency dance that can, at times, be less than fun.

---

All that said, thanks so much for the quick response!  And the update!  I have confirmed it's no longer throwing up that error, which is nice.  Also lets me play with the latest versions. :-)  So appreciate that.
